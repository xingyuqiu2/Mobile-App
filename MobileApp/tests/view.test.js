/* eslint-disable react/jsx-filename-extension */
// __tests__/app-test.js
import 'react-native';
import React from 'react';
import { render } from '@testing-library/react';
import {
  ProfileScreen, RepositoriesScreen, FollowingScreen, FollowersScreen,
} from '../view/app';

test('Matches default profile screen snapshot', () => {
  const { asFragment } = render(<ProfileScreen />);
  expect(asFragment()).toMatchSnapshot();
});

test('Matches default repositories screen snapshot', () => {
  const { asFragment } = render(<RepositoriesScreen />);
  expect(asFragment()).toMatchSnapshot();
});

test('Matches default following screen snapshot', () => {
  const { asFragment } = render(<FollowingScreen />);
  expect(asFragment()).toMatchSnapshot();
});

test('Matches default followers screen snapshot', () => {
  const { asFragment } = render(<FollowersScreen />);
  expect(asFragment()).toMatchSnapshot();
});
