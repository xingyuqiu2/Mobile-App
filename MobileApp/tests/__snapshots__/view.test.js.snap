// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches default followers screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
      data="[object Object]"
      initialnumtorender="10"
      maxtorenderperbatch="10"
      scrolleventthrottle="50"
      stickyheaderindices=""
      updatecellsbatchingperiod="50"
      viewabilityconfigcallbackpairs=""
      windowsize="21"
    >
      <view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
          >
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Name:Loading
            </text>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Username:Loading
            </text>
          </rctsafeareaview>
        </view>
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches default following screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
      data="[object Object]"
      initialnumtorender="10"
      maxtorenderperbatch="10"
      scrolleventthrottle="50"
      stickyheaderindices=""
      updatecellsbatchingperiod="50"
      viewabilityconfigcallbackpairs=""
      windowsize="21"
    >
      <view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
          >
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Name:Loading
            </text>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Username:Loading
            </text>
          </rctsafeareaview>
        </view>
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches default profile screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
    >
      <view>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Name
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Username: Loading
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Bio: Loading
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Website: Loading
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Email: Loading
        </text>
        <view
          focusable="true"
          style="opacity: 1;"
        >
          <text
            style="color: rgb(56, 41, 215); font-size: 18px;"
          >
            x repos
          </text>
        </view>
        <view
          focusable="true"
          style="opacity: 1;"
        >
          <text
            style="color: rgb(56, 41, 215); font-size: 18px;"
          >
            Followers: x
          </text>
        </view>
        <view
          focusable="true"
          style="opacity: 1;"
        >
          <text
            style="color: rgb(56, 41, 215); font-size: 18px;"
          >
            Following: x
          </text>
        </view>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Creation date: Loading
        </text>
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches default repositories screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
      data="[object Object]"
      initialnumtorender="10"
      maxtorenderperbatch="10"
      scrolleventthrottle="50"
      stickyheaderindices=""
      updatecellsbatchingperiod="50"
      viewabilityconfigcallbackpairs=""
      windowsize="21"
    >
      <view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
          >
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Repo name:Loading
            </text>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Username:Loading
            </text>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Description:Loading
            </text>
          </rctsafeareaview>
        </view>
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;
